### PR TITLE
Framework migration of SCA module

### PR DESCRIPTION
**Pull request by mateocervilla**
Related to: https://github.com/wazuh/wazuh/issues/19087

## Description

This PR includes all the necessary changes to give support to the SCA migrated tests in the `wazuh/wazuh` repository. Those tests will use this new structure.

### Note:
In the `_match()` method from `BaseMonitor` class it was needed to append matches results to `callback_result` for cases when `accumulations` value is greater than 1

### CODE REVIEW

**Comment by tdrauncieal** [comment]
tests done.

**Review by tdrauncieal — APPROVED** [review]

File: src/wazuh_testing/modules/sca/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (C) 2015-2023, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+PREFIX = r'.*sca.*'
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/sca/__init__.py:5]: Add new line at the end of the file.

File: src/wazuh_testing/modules/sca/patterns.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        line (str): line string to check for match.
+    '''
+    match = re.match(CB_SCA_SCAN_EVENT, line)
+    if match:
+        if json.loads(match.group(1))['type'] == 'summary':
+            return json.loads(match.group(1))
+
+
+# Variables
+TEMP_FILE_PATH = '/tmp'
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/sca/patterns.py:50]: Add new line at the end of the file.

File: src/wazuh_testing/modules/sca/patterns.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+# Error Messages
+ERR_MSG_REGEX_ENGINE = "Did not receive the expected 'SCA will use '.*' engine to check the rules' event"
+ERR_MSG_ID_RESULTS = 'Expected sca_has_integrity result events not found'
+ERR_MSG_SCA_SUMMARY = 'Expected SCA Scan Summary type event not found.'
+
+
+# Callback functions
+def callback_scan_id_result(line):
```
Comment (TomasTurina) [inline src/wazuh_testing/modules/sca/patterns.py:28]: I would define these two functions in the tests, not in the framework.

File: src/wazuh_testing/tools/monitors/base_monitor.py [hunk 4]
```diff
@@ -55,6 +55,11 @@ def _match(self, message: str, callback: Callable) -> bool:
         result = callback(message)
 
         # Update match result only if it's not None (i.e., there was a match)
-        self.callback_result = result if result else self.callback_result
+        if result:
```
Comment (TomasTurina) [inline src/wazuh_testing/tools/monitors/base_monitor.py:58]: Have you tested that this change doesn't break any test already migrated?

File: src/wazuh_testing/utils/file.py [hunk 5]
```diff
@@ -311,3 +313,41 @@ def delete_files(files: list[Union[str, os.PathLike]]) -> None:
...
+                file_list.append(file)
+                copy(os.path.join(src_folder, file), dst_folder)
+                remove_file(os.path.join(src_folder, file))
+        else:
+            for file in files_to_move:
+                if os.path.isfile(os.path.join(src_folder, file)):
+                    file_list.append(file)
+                    copy(os.path.join(src_folder, file), dst_folder)
+                    remove_file(os.path.join(src_folder, file))
+    return file_list
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/file.py:353]: Add new line at the end of the file.

File: src/wazuh_testing/modules/sca/patterns.py (hunk 2 above)
Comment (TomasTurina) [inline src/wazuh_testing/modules/sca/patterns.py:50]: Also, I think this define should be in `src/wazuh_testing/constants/paths/__init__.py`.

File: src/wazuh_testing/utils/file.py [hunk 6]
```diff
@@ -311,3 +313,41 @@ def delete_files(files: list[Union[str, os.PathLike]]) -> None:
...
+    Copy file with metadata and ownership to a specific destination.
+
+    Args:
+        source (str): Source file path to copy.
+        destination (str): Destination file.
+    """
+    shutil.copy2(source, destination)
+    source_stats = os.stat(source)
+
+    if sys.platform != 'win32':
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/file.py:329]: Use `platforms.WINDOWS`.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: src/wazuh_testing/modules/sca/patterns.py (hunk 3 above)
Comment (TomasTurina) [reply src/wazuh_testing/modules/sca/patterns.py:28]: Or instead, can't you use `generate_callback` function or refactor it to allow this in a more generic way?

**Review by TomasTurina — CHANGES_REQUESTED** [review]

**Review by TomasTurina — APPROVED** [review]

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**